### PR TITLE
feat(skyrim-platform): add new enums

### PR DESCRIPTION
**Pull request by CarlosLeyvaAyala**
Sources:
- https://www.creationkit.com/index.php?title=GetType_-_Form
- https://www.creationkit.com/index.php?title=GetWeaponType_-_Weapon
- https://www.creationkit.com/index.php?title=GetEquippedItemType_-_Actor

Notice how WeaponType and EquippedItemType are similar but quite different, because they are used for different functions (I'll add a clarification for both enums in help files).

### CODE REVIEW

File: skyrim-platform/src/platform_se/codegen/convert-files/Definitions.txt [hunk 1]
```diff
@@ -494,6 +494,175 @@ export const enum DxScanCode {
     MouseWheelDown
 }
 
+export const enum FormType {
```
Comment (Pospelove) [inline skyrim-platform/src/platform_se/codegen/convert-files/Definitions.txt:497]: Please add a link to creationkit.com as a comment

File: skyrim-platform/src/platform_se/codegen/convert-files/Definitions.txt [hunk 2]
```diff
@@ -494,6 +494,175 @@ export const enum DxScanCode {
...
+    Tree = 38,
+    VoiceType = 98,
+    Water = 84,
+    Weapon = 41,
+    Weather = 54,
+    WordOfPower = 118,
+    WorldSpace = 71
+}
+
+export const enum WeaponType {
```
Comment (Pospelove) [inline skyrim-platform/src/platform_se/codegen/convert-files/Definitions.txt:635]: same

File: skyrim-platform/src/platform_se/codegen/convert-files/Definitions.txt [hunk 3]
```diff
@@ -494,6 +494,175 @@ export const enum DxScanCode {
...
+    Mace,
+    Greatsword,
+    Battleaxe = 6,
+    Warhammer = 6,
+    Bow,
+    Staff,
+    Crossbow
+}
+
+export const enum EquippedItemType {
```
Comment (Pospelove) [inline skyrim-platform/src/platform_se/codegen/convert-files/Definitions.txt:649]: and here :)

**Review by Pospelove — CHANGES_REQUESTED** [review]
1) I left a few comments on corresponding lines.
2) Please also add changelog here https://github.com/skyrim-multiplayer/skymp/tree/main/docs/release/dev
3) Please document enums defined by SP here https://github.com/skyrim-multiplayer/skymp/tree/main/docs/skyrim_platform

Feel free to document as you want. It's ok if the changelog entry will be similar to docs.

See https://github.com/skyrim-multiplayer/skymp/commit/c9685c67ced38252d75d5ca0061a9974495825c0 for reference. The changelog entries (split into files) will be collected into SP 2.2 changelog like this https://github.com/skyrim-multiplayer/skymp/blob/main/docs/release/sp-2.1.md

File: skyrim-platform/src/platform_se/codegen/convert-files/Definitions.txt (hunk 3 above)
Comment (CarlosLeyvaAyala) [reply skyrim-platform/src/platform_se/codegen/convert-files/Definitions.txt:649]: Cool. I was thinking about that, but didn't know if that was acceptable.

I'll do it.

**Comment by CarlosLeyvaAyala** [comment]
> 1. I left a few comments on corresponding lines.
> 2. Please also add changelog here https://github.com/skyrim-multiplayer/skymp/tree/main/docs/release/dev
> 3. Please document enums defined by SP here https://github.com/skyrim-multiplayer/skymp/tree/main/docs/skyrim_platform
> 
> Feel free to document as you want. It's ok if the changelog entry will be similar to docs.
> 
> See [c9685c6](https://github.com/skyrim-multiplayer/skymp/commit/c9685c67ced38252d75d5ca0061a9974495825c0) for reference. The changelog entries (split into files) will be collected into SP 2.2 changelog like this https://github.com/skyrim-multiplayer/skymp/blob/main/docs/release/sp-2.1.md

Yep. I'm working on documenting these enums.
I'll post it as a draft later.

**Comment by Pospelove** [comment]
@CarlosLeyvaAyala ok, so the changelog is the only issue then.

File: skyrim-platform/src/platform_se/codegen/convert-files/Definitions.txt (hunk 1 above)
Comment (CarlosLeyvaAyala) [reply skyrim-platform/src/platform_se/codegen/convert-files/Definitions.txt:497]: I usually add comments as TSDoc.

Is that acceptable?

For example:
```ts
/** * Values were taken from {@link https://www.creationkit.com/index.php?title=GetType_-_Form} */
```

Comment (Pospelove) [reply skyrim-platform/src/platform_se/codegen/convert-files/Definitions.txt:497]: Yes

File: docs/skyrim_platform/new_types.md [hunk 4]
```diff
@@ -19,3 +19,61 @@ SkyrimPlatform currently only adds one type: `TESModPlatform`. Instances of this
...
+
+Here's a list of enum values declared in `skyrimPlatform.ts`.
+
+- `DxScanCode`. Key mappings from [Input Script][].
+
+  ```ts
+  Input.isKeyPressed(DxScanCode.F2);
+  ```
+
+<!-- TODO: Uncomment when these enums are actually published -->
```
Comment (Pospelove) [inline docs/skyrim_platform/new_types.md:35]: The irony is that they land with the same PR. So we need to uncomment before this PR can be accepted :)

**Review by Pospelove — CHANGES_REQUESTED** [review]

File: docs/skyrim_platform/new_types.md (hunk 4 above)
Comment (CarlosLeyvaAyala) [reply docs/skyrim_platform/new_types.md]: Done.

**Review by Pospelove — APPROVED** [review]

**Pull request merged**